### PR TITLE
[AD-550] no longer treat schema and catalog as patterns in getPrimaryKeys and getImportedKeys

### PR DESCRIPTION
**Pull request by alinaliBQ**
### Summary

make JDBC no longer treat schema and catalog as patterns in getPrimaryKeys and getImportedKeys

### Description

- remove testPrimaryKeyWildcards test
- make JDBC not treat schema and catalog as patterns in getPrimaryKeys and getImportedKeys
- add unit test to ensure that empty resultsets are returned when white space is passed as schema name to function getPrimaryKeys or getImportedKeys
### Related Issue

https://bitquill.atlassian.net/browse/AD-550

### Additional Reviewers
@affonsoBQ
@alinaliBQ
@andiem-bq
@alexey-temnikov
@birschick-bq
@mitchell-elholm
@RoyZhang2022

### CODE REVIEW

**Review by affonsov — DISMISSED** [review]
LGTM

File: src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java [hunk 1]
```diff
@@ -498,9 +498,9 @@ public ResultSet getPrimaryKeys(final String catalog, final String schema,
         final List<List<Object>> metaData = new ArrayList<>();
         final String regexSchemaPattern = convertPatternToRegex(schema);
         final String regexTableSchemaPattern = convertPatternToRegex(table);
-        if (schema == null || properties.getDatabase().matches(regexSchemaPattern)) {
+        if (isNullOrWhitespace(schema) || properties.getDatabase().matches(regexSchemaPattern)) {
```
Comment (mitchell-elholm) [inline src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java]: I might be missing context here but as I understand it, null and whitespace don't mean the same thing in getPrimaryKeys/getImportedKeys, `""` means without a schema whereas `null` means don't use the schema to narrow search. Not sure if I've understood this correctly but it seems that these should do slightly different things.  [reference](https://docs.oracle.com/javase/7/docs/api/java/sql/DatabaseMetaData.html#getImportedKeys(java.lang.String,%20java.lang.String,%20java.lang.String))

Comment (alinaliBQ) [reply src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java]: yea `""` and null do mean different things. The reason I made the change is that ODBC can only pass `""` to the JNI call and cannot pass `null` to my understanding, so I'm just enabling JDBC to send non-empty result sets. So basically when we pass `""` from ODBC we are trying to mean the same thing as `null`. I think since we implemented `getColumns` and `getTables` by treating `""` the same as `null` for schema and catalog, it should be okay?

Comment (mitchell-elholm) [reply src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java]: That seems reasonable

Comment (affonsov) [reply src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java]: this might break the behaviour of importedkeys tho, maybe you will need to have other API just for ODBC

Comment (affonsov) [reply src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java]: just because we need the different results and differentiate when is null or is empty
https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqlprimarykeys-function?view=sql-server-ver15

**Review by affonsov — COMMENTED** [review]
maybe will need a different implementation

**Comment by alinaliBQ** [comment]
> maybe will need a different implementation

if we're using different implementation for getImportedKeys / getPrimaryKeys then we might also need a different implementation for getColumns and getTables?

File: src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java (hunk 1 above)
Comment (alinaliBQ) [reply src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java]: I have instead reverted my code changes for getPrimaryKeys and getImportedKeys. I have updated the tests to make sure the behavior from JDBC is correct when empty strings are passed to getPrimaryKeys and getImportedKeys.

**Comment by affonsov** [comment]
The description needs to be updated

**Comment by alinaliBQ** [comment]
> The description needs to be updated

@affonsoBQ  I have updated. Please take a look

**Review by affonsov — APPROVED** [review]

File: src/test/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaDataTest.java [hunk 2]
```diff
@@ -499,6 +490,13 @@ void testGetPrimaryKeysArray() throws SQLException {
         Assertions.assertFalse(arrayPrimaryKeys.next());
     }
 
+    @Test
+    @DisplayName("Tests primary keys of array virtual tables with white space parameters. An empty ResultSet should be returned.")
```
Comment (RoyZhang2022) [inline src/test/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaDataTest.java]: "white space" should be empty string.

**Review by mitchell-elholm — APPROVED** [review]

File: src/test/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaDataTest.java (hunk 2 above)
Comment (alinaliBQ) [reply src/test/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaDataTest.java]: fixed
